tests/grid_test.py

```diff
@@ -0,0 +1,214 @@
+import numpy
+import pytest
+
+from smolyay.basis import ChebyshevFirstKind
+from smolyay.grid import (IndexGridGenerator, SmolyakGridGenerator,
+                          TensorGridGenerator,
+                          IndexGrid, NestedIndexGrid, generate_compositions)
+
+
+def test_error_basis_set_type():
+    """Test if raise TypeError works for IndexGridGenerator."""
+    basis_set = "not a BasisFunctionSet"
+    with pytest.raises(TypeError):
+        IndexGridGenerator(basis_set)
+
+
+def test_error_nested_basis_set_type():
+    """Test if raise TypeError works for SmolyakGridGenerator."""
+    basis_set = "not a NestedBasisFunctionSet"
```

Comment:
It would be a better test to use a regular `BasisFunctionSet` here rather than a string, as that would be a potential common mistake.